### PR TITLE
update pipelines

### PR DESCRIPTION
**Pull request by finswimmer**
* add python3.10 to release pipeline

* update actions/setup-python to @v2
* add python3.9 and python3.10 to checks
* update pytest for python >=3.6 otherwise it will fail on python3.10

### CODE REVIEW

File: make-nix-release.sh [hunk 1]
```diff
@@ -14,6 +14,7 @@ if [ "$OSTYPE" == "linux-gnu" ]; then
   $PYTHON get-poetry.py -y
   POETRY="$PYTHON $HOME/.poetry/bin/poetry"
   RUNTIMES[5]="${PYTHON39:+-P "3.9:$PYTHON39"}"
+  RUNTIMES[5]="${PYTHON310:+-P "3.10:$PYTHON310"}"
```
Comment (abn) [inline make-nix-release.sh:17]:
```suggestion
  RUNTIMES[6]="${PYTHON310:+-P "3.10:$PYTHON310"}"
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -22,12 +22,12 @@ jobs:
     strategy:
       matrix:
         os: [Ubuntu, MacOS, Windows]
-        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
+        python-version: [2.7, 3.5, 3.6, 3.7, 3.8, 3.9, "3.10"]
```
Comment (abn) [inline .github/workflows/main.yml:25]: Just for consistency, I would recommend we quote all values now.

**Review by abn — CHANGES_REQUESTED** [review]

File: make-nix-release.sh (hunk 1 above)
Comment (finswimmer) [reply make-nix-release.sh]: :+1:

File: .github/workflows/main.yml (hunk 2 above)
Comment (finswimmer) [reply .github/workflows/main.yml]: done

**Review by abn — APPROVED** [review]

**Pull request merged**